flax/linen/fp8_ops.py

```diff
@@ -130,141 +126,251 @@ def compute_amax_history(x, amax_history):
   new_history = jnp.roll(amax_history, shift=-1, axis=0).at[0].set(amax_update)
   return new_history
 
+def compute_scale(q_dtype, scale, amax_history):
+  dtype_max = get_fp8_max(q_dtype, jnp.float32)
+  amax_from_history = jnp.max(amax_history, axis=0)
+  new_scale = get_new_scale(amax_from_history, scale, dtype_max)
+  return new_scale
+
 
-def qdq_and_return(x, q_dtype, scale, amax_history, compute_dtype):
+def _compute_new_meta(x, q_dtype, scale, amax_history, compute_dtype):
   is_fm32 = scale.dtype == fm32 and amax_history.dtype == fm32
   # convert fm32->f32 so we can do math
   if is_fm32:
     amax_history = lax.convert_element_type(amax_history, jnp.float32)
     scale = lax.convert_element_type(scale, jnp.float32)
 
-  dtype_max = get_fp8_max(q_dtype, jnp.float32)
-  amax_from_history = jnp.max(amax_history, axis=0)
-  new_scale = compute_scale(amax_from_history, scale, dtype_max)
-
-  qx = quantize_dequantize(x, q_dtype, new_scale, compute_dtype)
-
+  new_scale = compute_scale(q_dtype, scale, amax_history)
```

Comment:
Overall, the changes look good to me. One more thing about the backward compatibility when migrating to the direct quantization. Do you think we should propose a new op `Fp8DirectDotGeneralOp`, like:
```python
class Fp8DirectDotGeneralOp(Fp8DotGeneralOp):
  def __call__(self, *args, **kwargs):
    ...
```